### ISSUE TITLE
Printing dependency graphs

### ISSUE THREAD

**Issue by wkoswara** [ssannandeji/Zenject-2019#361]
Hi, 

Is there a way to print the entire dependency graph? If one is not available already, where would one start in the code to be able to produce one?

**Comment by tertle** [comment]
Probably want to look at DiContainer.GetDependencyContracts 
From the readme:

DiContainer.GetDependencyContracts - Returns a list of all the types that the given type depends on. This might be useful, for exmaple, if you wanted to do some static analysis of your project, or if you wanted to automatically generate a dependency diagram, etc.

side note: example is spelt wrong on the readme

**Comment by svermeulen** [comment]
Check out `ObjectGraphVisualizer.CalculateObjectGraph`.  That code isn't actually used currently, but probably would help get you started.  There was a point awhile ago where zenject could actually generate images that showed the complete dependency graph, but that feature was removed because we didn't have time to maintain it properly.  But the code is still there when/if we decide to resurrect it

**Comment by pbwkoswara** [comment]
Thanks @tertle and @svermeulen for the suggestions. Haven't had a chance to take a look at those solutions yet, but will close this out when I do. 

Is there any plans currently to support this officially?

**Comment by wkoswara** [comment]
Thanks @tertle and @svermeulen for the suggestions. Haven't had a chance to take a look at those solutions yet, but will close this out when I do.

Is there any plans currently to support this officially?

**Comment by wkoswara** [comment]
Had to make some modifications in the Zenject code to get this to work, but it did. Thanks for the help.

**Comment by svermeulen** [comment]
When you say print you mean it prints the entire object graph to a file or something?  I'm just asking because I wonder if it would be useful for anyone else

**Comment by wkoswara** [comment]
Yes it prints the object graph to a file in DOT. I used ObjectGraphVisualizer.OutputObjectGraphToFile method. 

Sample code:
```
IEnumerable<Type> contractTypes = from t in Assembly.GetExecutingAssembly().GetTypes()
        where t.IsClass && t.Namespace == "Gourmet" && !t.IsAbstract
        select t;
ObjectGraphVisualizer.OutputObjectGraphToFile(m_diContainer, "output-graph", new List<Type>(), contractTypes);
```

I had to comment out a line of assertion code in ObjectGraphVisualizer.cs:
```
if (contractType.FullName.StartsWith("System.Collections.Generic.List"))
{
    var subTypes = contractType.GenericArguments();
    Assert.IsEqual(subTypes.Length, 1);

    var subType = subTypes[0];
    dependTypes = container.ResolveTypeAll(subType);
}
else
{
    dependTypes = container.ResolveTypeAll(contractType);
    // THIS ASSERTED ON VALID GRAPHS
    //Assert.That(dependTypes.Count <= 1);

```
